http-scraper/scraper.py

```diff
@@ -0,0 +1,293 @@
+                ms_wait=ms_wait,
+                data=data,
+            )
+
+        main_page_html = request_page_with_retry(
+            session=session,
+            url=base_url,
+            verification_text="ssSearchHyperlink",
+            logger=logger,
+            http_method=HTTPMethod.GET,
+            ms_wait=ms_wait,
+        )
+        main_soup = BeautifulSoup(main_page_html, "html.parser")
+        # build url for court calendar
+        search_page_id = None
+        for link in main_soup.select("a.ssSearchHyperlink"):
+            if court_calendar_link_text in link.text:
+                search_page_id = link["href"].split("?ID=")[1].split("'")[0]
+        if not search_page_id:
+            write_debug_and_quit(
+                verification_text="Court Calendar link",
+                page_text=main_page_html,
+                logger=logger,
+            )
+        search_url = base_url + "Search.aspx?ID=" + search_page_id
+
+    # hit the search page to gather initial data
+    search_page_html = request_page_with_retry(
+        session=session,
+        url=search_url
+        if odyssey_version < 2017
+        else urllib.parse.urljoin(base_url, "Home/Dashboard/26"),
+        verification_text="Court Calendar"
+        if odyssey_version < 2017
+        else "SearchCriteria.SelectedCourt",
+        http_method=HTTPMethod.GET,
+        logger=logger,
+        ms_wait=ms_wait,
+    )
+    search_soup = BeautifulSoup(search_page_html, "html.parser")
+
+    # we need these hidden values to POST a search
+    hidden_values = {
+        hidden["name"]: hidden["value"]
+        for hidden in search_soup.select('input[type="hidden"]')
+        if hidden.has_attr("name")
+    }
+    # get nodedesc and nodeid information from main page location select box
+    if odyssey_version < 2017:
+        location_option = main_soup.findAll("option")[0]
+        logger.info(f"location: {location_option.text}")
+        hidden_values.update({"NodeDesc": location, "NodeID": location_option["value"]})
+    else:
+        hidden_values["SearchCriteria.SelectedCourt"] = hidden_values[
+            "Settings.DefaultLocation"
+        ]  # TODO: Search in default court. Might need to add further logic later to loop through courts.
+
+    # get a list of JOs to their IDs from the search page
+    judicial_officer_to_ID = {
+        option.text: option["value"]
+        for option in search_soup.select(
+            'select[labelname="Judicial Officer:"] > option'
+            if odyssey_version < 2017
+            else 'select[id="selHSJudicialOfficer"] > option'
+        )
+        if option.text
+    }
+    # if judicial_officers param is not specified, use all of them
+    if not judicial_officers:
+        judicial_officers = list(judicial_officer_to_ID.keys())
+
+    # initialize variables to time script and build a list of already scraped cases
+    START_TIME = time()
+    cached_case_list = [
```

Comment:
Think I'm going to change this to check whether a file is in the container, rather than locally